### DISCUSSION TITLE
Command to list dependencies

### DISCUSSION THREAD

**Discussion by ivan-pi** [fortran-lang/fpm#682] [Ideas]
I was wondering today if fpm needs a command to list dependencies. Something like the output of

```text
conda list
```
or 
```text
python -m pip list [options]
```

What do you think?

**Comment by milancurcic** [comment]
I'm occasionally curious about dependencies and like to look at them for no good reason. Is this otherwise useful for anything? The most immediate dependencies are clear from the manifest, but those deeper down the tree are not as easy to see. This capability seems like it would be necessary, at least internally, for [version constraint resolution](https://github.com/fortran-lang/fortran-lang.org/wiki/GSoC-2022-Project-ideas#version-constraint-resolution). It could be useful to be part of the UI as well.

**Reply by ivan-pi** [reply]
One reason I can think of is to debug version conflicts. In most of my fpm projects I'm currently just tracking the main branches. At some point this is bound to break with API changes. If my project breaks at a certain point, I can go to that specific project and back-track through the commit/version history

A second reason would be to inspect and avoid certain dependencies for whatever reason (poor code quality, proven danger, user trust...).

**Comment by ivan-pi** [comment]
Here is a preview output processed with Graphviz:


![test](https://github.com/user-attachments/assets/e6805415-33f3-40fa-a442-7ed74989dab5)
